### DISCUSSION TITLE
Use an Origin Access Control when S3 Buckets and Cloudfront Distribution are in Different Stacks

### DISCUSSION THREAD

**Discussion by scottbisker** [aws/aws-cdk#31469] [Q&A]
With the addition of OAC for Cloudfront, what's the process for utilizing OAC when the S3 Buckets reside in a resource stack and are dependencies of the cloudfront stack that contains the Cloudfront Distribution?  When attempting to switch to cloudfront_origins.S3BucketOrigin.with_origin_access_control, CDK fails to synthesize because of a cyclical dependency.  Presumably this is because the Cloudfront distribution ID doesn't exist prior to creation.  Has anyone come up with a solution to this problem?  It's quite a common scenario where  resources would be created in different stacks.

**Comment by khushail** [comment]
@scottbisker , this might be helpful - https://github.com/aws/aws-cdk/issues/31462#issuecomment-2356869786
